src/include/baselib/jni/JniEnvironment.h

```diff
@@ -136,6 +137,69 @@ namespace bl
                 return m_jniEnv -> GetVersion();
             }
 
+            jobjectRefType getObjectRefType( SAA_in const jobject object ) const NOEXCEPT
+            {
+                return m_jniEnv -> GetObjectRefType( object );
+            }
+
+            void deleteLocalRef( SAA_in const jobject object ) const NOEXCEPT
+            {
+                m_jniEnv -> DeleteLocalRef( object );
+            }
+
+            template
+            <
+                typename T
+            >
+            GlobalReference< T > createGlobalReference( SAA_in const LocalReference < T >& localReference ) const
+            {
+                auto globalReference = GlobalReference< T >::attach(
+                    reinterpret_cast< T >( m_jniEnv -> NewGlobalRef( localReference.get() ) )
+                    );
+
+                BL_CHK_T(
+                    nullptr,
+                    globalReference.get(),
+                    JavaException(),
+                    BL_MSG()
+                        << "Failed to create new global reference."
+                    );
+
+                return globalReference;
+            }
+
+            void deleteGlobalRef( SAA_in const jobject object ) const NOEXCEPT
+            {
+                m_jniEnv -> DeleteGlobalRef( object );
```

Comment:
can this fail? if so I think it should do EH and wrap it in BL_NOEXCEPT_* block...